### PR TITLE
webui: fix empty string values

### LINKED ISSUE #9: webui: empty values lead to faulty 'default.nix'
Passing empty or null values through JSON to `default.nix` can result in a faulty configuration:

<details>

<summary> default.nix </summary>

```nix
{
  pkgs,
  config,
  inputs,
  lib,
  ...
}: {
  homestakeros = {
    addons = {
      mev-boost = {
        enable = false;
        endpoint = "http://127.0.0.1:18550";
      };
    };
    consensus = {
      lighthouse = {
        dataDir = "/var/mnt/lighthouse";
        enable = false;
        endpoint = "http://127.0.0.1:5052";
        execEndpoint = "";
        jwtSecretFile = "";
        slasher = {
          enable = false;
          historyLength = 4096;
          maxDatabaseSize = 256;
        };
      };
    };
    execution = {
      erigon = {
        dataDir = "/var/mnt/erigon";
        enable = true;
        endpoint = "http://127.0.0.1:8551";
        jwtSecretFile = "";
      };
    };
    localization = {
      hostname = "foo";
      timezone = "";
    };
    mounts = {};
    ssh = {
      authorizedKeys = [];
      privateKeyFile = "";
    };
    vpn = {
      wireguard = {
        configFile = "";
        enable = false;
        interfaceName = "wg0";
      };
    };
  };
}
```

</details>

Options with empty or null values should be removed from the JSON since they can produce the following types of errors, even if building is possible without setting the `ssh.privateKeyFile` option:

```shell
[I] kari@torque ~/W/HomestakerOS (main) [SIGINT]> tail -f pipe | sh
tail: inotify resources exhausted
tail: inotify cannot be used, reverting to polling
warning: Git tree '/home/kari/Workspace/HomestakerOS' is dirty
error:
       … while calling the 'derivationStrict' builtin

         at /builtin/derivation.nix:9:12: (source not available)

       … while evaluating derivation 'kexec-tree'
         whose name attribute is located at /nix/store/s4jqyj35hii03rs7j5n6vn7gpgp6ja81-source/pkgs/stdenv/generic/make-derivation.nix:300:7

       … while evaluating attribute 'buildCommand' of derivation 'kexec-tree'

         at /nix/store/s4jqyj35hii03rs7j5n6vn7gpgp6ja81-source/pkgs/build-support/trivial-builders/default.nix:87:14:

           86|       enableParallelBuilding = true;
           87|       inherit buildCommand name;
             |              ^
           88|       passAsFile = [ "buildCommand" ]

       (stack trace truncated; use '--show-trace' to show the full trace)

       error: A definition for option `homestakeros.ssh.privateKeyFile' is not of type `null or path'. Definition values:
       - In `/nix/store/8kfdf8sl274jqd3x7n223axv8wg7vpxy-source/nixosConfigurations/foo': ""
```

### PR DESCRIPTION
**Pull request by jormeli**
Also fixes some formatting issues

### CODE REVIEW

File: webui/src/Components/ConfigurationForm.tsx [hunk 1]
```diff
@@ -272,8 +272,16 @@ const ConfigurationForm = () => {
     const result = recursiveReplace(structuredClone(schema))
     const formData = new FormData(e.target as HTMLFormElement)
     const formDataJson = Object.fromEntries(formData.entries())
-    Object.entries(formDataJson).forEach(([key, value]) => {
+    Object.entries(formDataJson).forEach(([key, _value]) => {
       const schemaEntry = jp.query(schema, key)
+      let value = _value
+      if (value === '') {
+        if (schemaEntry.length > 0 && schemaEntry[0]['type'] === 'nullOr') {
+          value = null
```
Comment (tupakkatapa) [inline webui/src/Components/ConfigurationForm.tsx:280]: Don't know much about the code but I get the following error when trying to run `yarn build`:
```shell
[kari@torque:~/Workspace/HomestakerOS]$ yarn build
yarn run v1.22.19
$ tsc -p webui/tsconfig.json && vite build webui
webui/src/Components/ConfigurationForm.tsx:280:11 - error TS2322: Type 'null' is not assignable to type 'FormDataEntryValue'.

280           value = null
              ~~~~~


Found 1 error in webui/src/Components/ConfigurationForm.tsx:280

error Command failed with exit code 2.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

Comment (jormeli) [reply webui/src/Components/ConfigurationForm.tsx:280]: my bad, too hasty

Comment (tupakkatapa) [reply webui/src/Components/ConfigurationForm.tsx:280]: no problem

Comment (jormeli) [reply webui/src/Components/ConfigurationForm.tsx]: that should work now

**Comment by tupakkatapa** [comment]
LGTM, values seem to be correct now and the host compiles

Not related to this PR, but just for the record -- there are some evaluation issues, at least with the following module options when they receiving a `null` value:

  - `execution.erigon.jwtSecretFile`
  - `consensus.lighthouse.jwtSecretFile`
  - `vpn.wireguard.configFile`
  
Handling of these should be fixed at [ponkila/nixobolus](https://github.com/ponkila/nixobolus).

**Comment by tupakkatapa** [comment]
This can be squashed and merged.

**Comment by jhvst** [comment]
There is also the point that some fields could be set as `required` from the UI level. But if Jesse knows how to handle this on Nix level, that's always better -- we currently assume that certain fields are set, although just because _we_ use them in our configuration, but in a pedantic sense there is no real reason why a Linux host would not work without a VPN, or erigon or lighthouse without the JWT token. In that sense it makes to do in Nix instead.

**Pull request merged**